sdk/calling-sdk.js

```diff
@@ -9,224 +8,234 @@ let call;
-                    callNotification.startTimer();
-                }
-            }
-        });
-    
-        call.on('remote_media', (track) => {
-            document.getElementById('customer-remote-audio').srcObject = new MediaStream([track]);
-        });
-    
-        call.on('disconnect', (correlationId) => {
-            closeCallWindow();
-        });
-    
-        call.dial(localAudioStream);
-    } catch (err) {
-        console.log("DEMO: Failed in initiating call");
-    }
+  try {
+    const destination = {
+      value: number,
+    };
+    openCallWindow(number);
+    await getMediaStreams();
+
+    // Create call object
+    call = line.makeCall({
+      type: "uri",
+      address: destination.value,
+    });
+
+    call.on("caller_id", (CallerIdEmitter) => {
+      updateCallerId(CallerIdEmitter);
+    });
+
+    call.on("progress", (correlationId) => {
+      // Add ringback on progress
+    });
+
+    call.on("connect", (correlationId) => {
+      if (number === "5007") {
+        secondCallNotification.startTimer();
+        secondCallNotification.enableCompleteTransfer();
+      } else {
+        if (window.location.href.includes("mytrips")) {
+          callNotification.startTimer();
+        }
+      }
+    });
+
+    call.on("remote_media", (track) => {
+      document.getElementById("customer-remote-audio").srcObject =
+        new MediaStream([track]);
+    });
+
+    call.on("disconnect", (correlationId) => {
+      closeCallWindow();
+    });
+
+    // Initiate an outbound call
+    call.dial(localAudioStream);
+  } catch (err) {
+    console.log("SDK: Failed in initiating call");
+  }
 }
 
 // Step 6: Fetch the call instance from the call notification, setup call listeners, create media stream and answer the incoming call
 async function answerCall() {
-    try {
-        fetchCallerBooking();
-
-        await getMediaStreams();
-
-        incomingCall.answer(localAudioStream);
-        callNotification.startTimer();
-
-        incomingCall.on('caller_id', (CallerIdEmitter) => {
-            callerName.innerText = 'Harvey Spector';
-            callerNumber.innerText = CallerIdEmitter.callerId.num;
-        });
-
-        incomingCall.on('remote_media', (track) => {
-            document.getElementById('agent-remote-audio').srcObject = new MediaStream([track]);
-        });
-
-
-        incomingCall.on('disconnect', (correlationId) => {
-            closeCallWindow();
-        });
-    } catch (err) {
-        console.log("DEMO: failed to answer the call.");
-    } 
+  try {
+    fetchCallerBooking();
+
+    await getMediaStreams();
+
+    //Accept incoming call
+    incomingCall.answer(localAudioStream);
+    callNotification.startTimer();
+
+    incomingCall.on("caller_id", (CallerIdEmitter) => {
+      callerName.innerText = "Harvey Spector";
+      callerNumber.innerText = CallerIdEmitter.callerId.num;
+    });
+
+    incomingCall.on("remote_media", (track) => {
```

Comment:
We also should add an event listener for hold success, hold error, resume success and resume failure for both the call objects in this file.